src/core/hle/D3D8/Direct3D9/CxbxPixelShaderTemplate.hlsl

```diff
@@ -334,6 +340,31 @@ float3 DoBumpEnv(const float4 TexCoord, const float4 BumpEnvMat, const float4 sr
 
 PS_OUTPUT main(const PS_INPUT xIn)
 {
+	//Fog
+	const float fogDepth      =   xIn.iFog.x; // Don't abs this value! Test-case : DolphinClassic xdk sampl
+	const float fogTableMode  =   FOGINFO.x;
+	const float fogDensity    =   FOGINFO.y;
+	const float fogStart      =   FOGINFO.z;
+	const float fogEnd        =   FOGINFO.w;  
+
+	const int FOG_TABLE_NONE    = 0;
+	const int FOG_TABLE_EXP     = 1;
+	const int FOG_TABLE_EXP2    = 2;
+	const int FOG_TABLE_LINEAR  = 3;
+
+    float fogFactor;
+
+    if(fogTableMode == FOG_TABLE_NONE) 
+       fogFactor = fogDepth;
+    if(fogTableMode == FOG_TABLE_EXP) 
+       fogFactor = 1 / exp(fogDepth * fogDensity); // 1 / e^(d * density)
+    if(fogTableMode == FOG_TABLE_EXP2) 
+       fogFactor = 1 / exp(pow(fogDepth * fogDensity, 2)); // 1 / e^((d * density)^2)
+    if(fogTableMode == FOG_TABLE_LINEAR) 
+       fogFactor = (fogEnd - fogDepth) / (fogEnd - fogStart);
+	if (FOGENABLE == 0)
+	   fogFactor = 1;
+	   
```

Comment:
I am not sure how flow control in HLSL should be. could this be revised to    if(fogTableMode == FOG_TABLE_NONE) 
       fogFactor = fogDepth;
	if (FOGENABLE == 0)
    {
	   fogFactor = 1;
    }
    else
    {
       if(fogTableMode == FOG_TABLE_EXP) 
          fogFactor = 1 / exp(fogDepth * fogDensity); // 1 / e^(d * density)
       if(fogTableMode == FOG_TABLE_EXP2) 
          fogFactor = 1 / exp(pow(fogDepth * fogDensity, 2)); // 1 / e^((d * density)^2)
       if(fogTableMode == FOG_TABLE_LINEAR) 
          fogFactor = (fogEnd - fogDepth) / (fogEnd - fogStart);
    }

Reply:
@medievil1 I am asking about flow control, not the formatting. did you try the flow control I listed above?

Reply:
changed and committed, though did not see any noticeable differences